Source/Core/Core/PatchEngine.cpp

```diff
@@ -211,6 +224,52 @@ static void ApplyPatches(const std::vector<Patch>& patches)
   }
 }
 
+std::unique_ptr<std::map<u64, std::vector<u8>>> CalculateDiscPatches(const DiscIO::Volume &disc) {
+  if (s_file_patches.empty())
+    return nullptr;
+
+  auto partition = disc.GetGamePartition();
+  const DiscIO::FileSystem *fs = disc.GetFileSystem(partition);
+
+  auto patches = std::make_unique<std::map<u64, std::vector<u8>>>();
+
+  for (const auto& patch : s_file_patches) {
+    if (!patch.active)
+      continue;
+
+    const auto& file = fs->FindFileInfo(patch.path);
+    if (file) {
+      for (const auto& entry : patch.entries) {
+        u64 disc_offset = disc.PartitionOffsetToRawOffset(file->GetOffset(), partition) + entry.address;
+        std::vector<u8> data;
+        switch (entry.type)
+        {
+          case PatchType::Patch32Bit:
+            data.push_back(static_cast<u8>(entry.value>>24));
+            data.push_back(static_cast<u8>(entry.value>>16));
```

Comment:
The relevant fallthrough cases should have a comment indicating it's intentional, making sure the paths aren't forgotten about when migration to C++17 is done (where `[[fallthrough]];` can be utilized)